### PR TITLE
Upgrade gpu build to CUDA 10 + cudnn 7.3

### PR DESCRIPTION
**Pull request by HectorSVC**
Upgrade gpu build to CUDA 10 + cudnn 7.3
Keep the cuda 9.1 in the separate buld pipeline

### CODE REVIEW

**Comment by raymondxyang** [comment]
Hi Hector could u also copy the new content in yaml to the py-packaging.yml? Thanks

File: tools/ci_build/github/linux/run_dockerbuild.sh [hunk 1]
```diff
@@ -25,7 +28,7 @@ echo "bo=$BUILD_OS bd=$BUILD_DEVICE bdir=$BUILD_DIR pv=$PYTHON_VER bex=$BUILD_EX
 
 cd $SCRIPT_DIR/docker
 if [ $BUILD_DEVICE = "gpu" ]; then
-    IMAGE="ubuntu16.04-cuda9.1-cudnn7.1"
+    IMAGE="ubuntu16.04-$CUDA_VER"
     docker build -t "onnxruntime-$IMAGE" --build-arg PYTHON_VERSION=${PYTHON_VER} -f Dockerfile.ubuntu_gpu .
```
Comment (HectorSVC) [inline tools/ci_build/github/linux/run_dockerbuild.sh]: >Dockerfile.ubuntu_gpu [](start = 85, length = 21)

should switch the docker file as well, will fix it for cuda9.1 build

File: tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml [hunk 2]
```diff
@@ -47,7 +47,7 @@ jobs:
       displayName: 'Run build script'
       inputs:
         scriptPath: 'tools/ci_build/github/linux/run_dockerbuild.sh'
-        args: '-c Release -o ubuntu16.04 -d gpu -r $(Build.BinariesDirectory) -p $(python.version) -x "--build_wheel --use_mkldnn"'
+        args: '-c Release -o ubuntu16.04 -d gpu -c cuda9.1-cudnn7.1 -r $(Build.BinariesDirectory) -p $(python.version) -x "--build_wheel --use_mkldnn"'
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml]: did you mean to say cuda10 here?

File: tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline-cuda9.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+jobs:
+- job: Linux_CI_GPU_Dev
+  pool: Linux-GPU-CUDA10
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline-cuda9.yml]: why is the pool name cuda10 when the cuda version used is 9.1? do we've just one pool?

File: tools/ci_build/github/azure-pipelines/win-gpu-ci-pipeline-cuda9.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+jobs:
+- job: Windows_CI_GPU_Dev
+  pool: Win-GPU-CUDA10
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/win-gpu-ci-pipeline-cuda9.yml]: same comment as above.

File: tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml (hunk 2 above)
Comment (HectorSVC) [reply tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml]: for the python package I think we are still releasing with cuda 9.1 for Azure?

---
In reply to: [239301044](https://github.com/Microsoft/onnxruntime/pull/112#discussion_r239301044) [](ancestors = 239301044)

File: tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline-cuda9.yml (hunk 3 above)
Comment (HectorSVC) [reply tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline-cuda9.yml]: this is the build agent pool name. We created separate pool to make the switch smooth

---
In reply to: [239301328](https://github.com/Microsoft/onnxruntime/pull/112#discussion_r239301328) [](ancestors = 239301328)

File: tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml (hunk 2 above)
Comment (pranavsharma) [reply tools/ci_build/github/azure-pipelines/azure-pipelines-py-packaging.yml]: yeah, not sure if Azure upgraded to cuda 10. @vinitra might know.

**Pull request merged**